spec/furniture/marketplace/marketplace_spec.rb

```diff
@@ -4,4 +4,16 @@
   it { is_expected.to have_many(:products).inverse_of(:marketplace).dependent(:destroy) }
   it { is_expected.to have_many(:orders).inverse_of(:marketplace) }
   it { is_expected.to have_many(:carts).inverse_of(:marketplace).dependent(:destroy) }
+
+  describe "#destroy" do
+    subject(:destroy) { -> { marketplace.destroy } }
+
+    let(:marketplace) { create(:marketplace, :with_orders) }
+    let(:orders) { marketplace.orders }
+
+    context "when there are `Order`s" do
+      it { is_expected.not_to raise_error }
+      it { is_expected.not_to change(orders, :count) }
```

Comment:
@anaulin - does this make u the happiest you've ever been?

Reply:
It does!